devices/MS5611/Ms5611.cs

```diff
@@ -0,0 +1,201 @@
+        }
+
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="sampling">Sampling Mode</param>
+        public void SetSampling(Sampling sampling)
+        {
+            _sampling = sampling;
+            _delayForSampling = sampling switch
+            {
+                Sampling.UltraLowPower => 1,
+                Sampling.LowPower => 2,
+                Sampling.Standard => 3,
+                Sampling.HighResolution => 5,
+                Sampling.UltraHighResolution => 10,
+                _ => 5
+            };
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        [Telemetry("Temperature")]
+        public Temperature ReadTemperature() => Temperature.FromDegreesCelsius(ReadTrueTemperature());
+
+        /// <summary>
+        ///  Reads the pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Atmospheric pressure
+        /// </returns>
+        [Telemetry("Pressure")]
+        public Pressure ReadPressure() => Pressure.FromMillibars(ReadTruePressure(compensation: false));
+
+        /// <summary>
+        ///  Calculates the pressure at sea level when given a known altitude
+        /// </summary>
+        /// <param name="altitude" >
+        ///  Altitude in meters
+        /// </param>
+        /// <returns>
+        ///  Pressure
+        /// </returns>
+        public Pressure ReadSeaLevelPressure(Length altitude) => WeatherHelper.CalculateSeaLevelPressure(ReadPressure(), altitude, ReadTemperature());
+
+        /// <summary>
+        ///  Reads raw temperature value from the sensor
+        /// </summary>
+        /// <param name="compensation" >
+        ///  Flag indicating if compensation calculation for low temperatures (lower than 20 degrees C) should be done
+        /// </param>
+        /// <returns>
+        ///  Temperature in Celsius
+        /// </returns>
+        public double ReadTrueTemperature(bool compensation = false)
+
+        {
+            long rawTemp = ReadRawData((byte)((byte)CommandAddresses.SamplingRateTemperature + _sampling));
+            long diffActualAndReference = rawTemp - _calibrationData.ReferenceTemperature * 256;
+            double actualTemperature = 2000 + (double)diffActualAndReference * _calibrationData.TemperatureCoefficientOfTheTemperature / 8388608;
+            double compensationValue = 0;
+            if (compensation)
+            {
+                if (actualTemperature < 2000)
+                {
+                    compensationValue = diffActualAndReference * diffActualAndReference / (2 << 30);
+                }
+            }
+            var temp = actualTemperature - compensationValue;
+            return (double)temp / 100;
+        }
+
+        internal int ReadRegister(int numberOfByte)
```

Comment:
See some of the other comments, for this one use directly the enum CommandAddresses  and rename the numberOfByte by registerAddress. And it will be much cleaner and easier to maintain as well